Libraries/Components/ScrollView/ScrollView.js

```diff
@@ -404,7 +404,34 @@ const ScrollView = React.createClass({
   },
 
   /**
-   * Deprecated, do not use.
+   * If this is a vertical ScrollView scrolls to the bottom.
+   * If this is a horizontal ScrollView scrolls to the right.
+   *
+   * Use `scrollToEnd()` for immediate scrolling,
+   * `scrollToEnd({animated: true})` for animated scrolling.
+   *
+   * See `ScrollView#scrollToEnd`.
+   */
+  scrollToEnd: function(
+    options?: { animated?: boolean },
+  ) {
+    if (Platform.OS === 'ios') {
+      this.getScrollResponder().scrollResponderScrollToEnd(options);
+    } else if (Platform.OS === 'android') {
+      // On Android scrolling past the end of the ScrollView gets clipped
+      // - scrolls to the end.
+      if (this.props.horizontal) {
+        this.scrollTo({x: 1000*1000, animated: options && options.animated});
+      } else {
+        this.scrollTo({y: 1000*1000, animated: options && options.animated});
+      }
```

Comment:
The advantage of doing this is we don't need to add and maintain any native code - the Android ScrollView essentially supports scrolling to the end using the native method `scrollTo` that's already implemented.